core/storage/blog_statistics/gae_models_test.py

```diff
@@ -0,0 +1,817 @@
+            blog_stats_models.BlogPostViewsAggregatedStatsModel
+            .get_deletion_policy(),
+            base_models.DELETION_POLICY.NOT_APPLICABLE
+        )
+
+    def test_create_and_get_stats_models(self) -> None:
+        blog_stats_models.BlogPostViewsAggregatedStatsModel.create(
+            self.BLOG_POST_TWO_ID
+        )
+        stats_model = blog_stats_models.BlogPostViewsAggregatedStatsModel.get(
+            self.BLOG_POST_TWO_ID)
+
+        self.assertEqual(stats_model.id, self.BLOG_POST_TWO_ID)
+
+        # Test should raise exception if model with given blog post id already
+        # exists.
+        with self.assertRaisesRegex(
+            Exception,
+            'A blog post views stats model with the given blog post ID'
+            'exists already.'):
+            blog_stats_models.BlogPostViewsAggregatedStatsModel.create(
+                 self.BLOG_POST_TWO_ID
+            )
+
+    def test_get_model_association_to_user(self) -> None:
+        self.assertEqual(
+            blog_stats_models.BlogPostViewsAggregatedStatsModel
+                .get_model_association_to_user(),
+            base_models.MODEL_ASSOCIATION_TO_USER.NOT_CORRESPONDING_TO_USER
+        )
+
+    def test_get_export_policy(self) -> None:
+        expected_export_policy_dict = {
+            'views_by_hour': base_models.EXPORT_POLICY.NOT_APPLICABLE,
+            'views_by_date': base_models.EXPORT_POLICY.NOT_APPLICABLE,
+            'views_by_month': base_models.EXPORT_POLICY.NOT_APPLICABLE,
+            'created_on': base_models.EXPORT_POLICY.NOT_APPLICABLE,
+            'last_updated': base_models.EXPORT_POLICY.NOT_APPLICABLE,
+            'deleted': base_models.EXPORT_POLICY.NOT_APPLICABLE
+        }
+        self.assertEqual(
+            blog_stats_models.BlogPostViewsAggregatedStatsModel
+                .get_export_policy(),
+            expected_export_policy_dict
+        )
+
+
+class BlogPostReadsAggregatedStatsModelUnitTests(test_utils.GenericTestBase):
+    """Test the BlogPostReadsAggregatedStatsModel class."""
+
+    BLOG_POST_ONE_ID: Final = 'blog_post_one'
+    BLOG_POST_TWO_ID: Final = 'blog_post_two'
+
+    def setUp(self) -> None:
+        """Set up blog post reads aggregated stats models in datastore for use
+        in testing.
+        """
+        super().setUp()
+
+        self.model_one = (
+            blog_stats_models.BlogPostReadsAggregatedStatsModel(
+                id=self.BLOG_POST_ONE_ID,
+                reads_by_hour={},
+                reads_by_date={},
+                reads_by_month={}
+            )
+        )
+        self.model_one.update_timestamps()
+        self.model_one.put()
+
+    def test_get_deletion_policy(self) -> None:
+        self.assertEqual(
+            blog_stats_models.BlogPostReadsAggregatedStatsModel
+            .get_deletion_policy(),
+            base_models.DELETION_POLICY.NOT_APPLICABLE
+        )
+
+    def test_create_and_get_stats_models(self) -> None:
```

Comment:
Test names should describe the action being taken and the outcome of the test instead of just the method being tested. This would not be okay in production code but the extra verbosity helps understand the behavior being tested just by its name. For e.g. in this case, the full test name can be called `BlogPostReadsAggregatedStatsModelUnitTests.test_create_raises_exception_with_existing_blog_id`. 

Also, I'm not completely sure but this test might be testing multiple behaviors that woud ideally each have their own test.

Same comments as above for all the tests.

Reply:
DOne

Reply:
Thanks for taking care of this change